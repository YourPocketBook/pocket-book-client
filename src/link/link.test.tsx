import { fireEvent, render } from "@testing-library/react";
import React from "react";
import ReactDOM from "react-dom";
import { Link } from ".";
import { history } from "../history";

describe("Home Page", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Link href="/test">Test</Link>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly", () => {
    const { container } = render(<Link href="/test">Test</Link>);

    expect(container).toMatchSnapshot();
  });

  it("pushes to history when clicked", () => {
    history.push = jest.fn();

    const target = {
      pathname: "/test",
      search: "?testid=42"
    };

    const { getByText } = render(<Link href="/test?testid=42">Test</Link>);
    fireEvent.click(getByText("Test"), {
      target
    });

    expect(history.push).toHaveBeenCalledWith(target);
  });
});
