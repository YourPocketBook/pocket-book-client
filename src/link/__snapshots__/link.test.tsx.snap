// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders correctly 1`] = `
<div>
  <a
    class=""
    href="/test"
  >
    Test
  </a>
</div>
`;
